redfish_client/resource.py

```diff
@@ -109,6 +119,41 @@ def execute_action(self, action_name, payload):
             return self._connector.post(action.target, payload=payload)
         raise KeyError("Action with {} does not exist".format(action_name))
 
+    def wait_for_value_at(self, expected, *stat, fails=None, poll_interval=3, timeout=30*60):
+        """
+        :param stat: single ker or dot-syntax key
+        :param expected: expected value
+        :param fails: list or tuple of fail values
+        :param poll_interval: number of requests per second
+        :param timeout: timeout in seconds
+        :return: actual value if match is found
+        Raises:
+            MissingDataIdException: If '@odata.id' is missing in the content
+            TimedOutError: If timeout is exceeded.
+            FailedError: If value matches one of the fail values
+        """
+
+        if "@odata.id" not in self._content:
+            raise MissingDataIdException("Element does not have '@odata.id' attribute, cannot wait for a stat inside "
+                                         "inner object")
+
+        start_time = time.time()
+
+        while time.time() <= start_time + timeout:
+            self._refresh_cache()
+
+            actual_value = reduce(operator.getitem, stat, self._content)
+            if hasattr(actual_value, "raw"):
```

Comment:
It is expected to return `False` in some cases, for example when we wait for `PowerState` stat, the value is not returned as `Resource` instance but is returned as `str`, in this case `.raw` is not called.

Reply:
The `hasattr` call in the condition will always return `False` here, since we are digging into raw content. So the whole `if` clause can be safely removed.